scripts/server.py

```diff
@@ -550,66 +567,544 @@ def handle_client(conn, addr):
+                            flush=True)
+                    except requests.exceptions.ConnectTimeout as conn_timeout_exec:
+                        # Check if were still able to retry
+                        if submission_retries < bw_request_max_retries:
+                            can_retry = True
+                            submission_retries += 1
+                            # Calculate the backoff time before making another request
+                            request_backoff_time = bw_request_backoff_factor * (2 ** (submission_retries - 1))
+
+                            if debug_birdweather_submissions:
+                                print(
+                                    f"BirdWeather Submission Error:: Detection POST - Connection Timeout! Retrying after {request_backoff_time}s, submission_retries:{submission_retries}, can_retry:{can_retry}, timeout_set:{bw_default_request_timeout} - {conn_timeout_exec}",
+                                    flush=True)
+                            else:
+                                print(
+                                    f"BirdWeather Submission Error:: Detection POST - Connection Timeout! Retrying after {request_backoff_time}s, Retry Attempt:{submission_retries} - {conn_timeout_exec}",
+                                    flush=True)
+                        else:
+                            # Unable to retry anymore
+                            can_retry = False
+                            print(
+                                f"BirdWeather Submission Error:: Detection POST - Connection Timeout! Max retries of {bw_request_max_retries} reached - {conn_timeout_exec}",
+                                flush=True)
+                    except requests.exceptions.ConnectionError as conn_err_exec:
+                        if submission_retries < bw_request_max_retries:
+                            can_retry = True
+                            submission_retries += 1
+                            # Calculate the backoff time before making another request
+                            request_backoff_time = bw_request_backoff_factor * (2 ** (submission_retries - 1))
+
+                            print(
+                                f"BirdWeather Submission Error:: Soundscape POST - Connection Error! Retrying after {request_backoff_time}s, Retry Attempt:{submission_retries} - {conn_err_exec}",
+                                flush=True)
+                        else:
+                            print(
+                                f"BirdWeather Submission Error:: Soundscape POST - Connection Error! Not Retrying - {conn_err_exec}",
+                                flush=True)
+                    except requests.exceptions.RequestException as request_exc:
+                        # Check if the status code is one that we can try on
+                        req_status_code = request_exc.response.status_code
```

Comment:
Is this not detection_response_status_code? Isn't this already covered above?

Reply:
This is just checking the response code once we're in exception, it's separate from the check in the try body for 2XX status codes.. which isn't required